lib/src/transaction.rs

```diff
@@ -147,15 +147,9 @@ pub fn create_op_metadata(
     }
 }
 
-struct NewRepoData {
-    operation: Operation,
-    view: View,
-    index: Box<dyn ReadonlyIndex>,
-}
-
 pub struct UnpublishedOperation {
     repo_loader: RepoLoader,
-    data: NewRepoData,
+    repo: Arc<ReadonlyRepo>,
```

Comment:
nit: might be better to extract `OpHeadsStore` upfront (because we no longer need full access to `repo_loader`.

Reply:
Seems reasonable to me, although it probably doesn't matter much. I added a new commit to do this :)